contracts/VestedPayment.sol

```diff
@@ -0,0 +1,85 @@
+pragma solidity ^0.4.13;
+
+import "zeppelin-solidity/contracts/math/SafeMath.sol";
+import "zeppelin-solidity/contracts/ownership/Ownable.sol";
+import "./LifToken.sol";
+
+contract VestedPayment is Ownable {
+  using SafeMath for uint256;
+
+  // when the vested schedule starts
+  uint256 public startTime;
+
+  // how much each period will last
+  uint256 public timePerPeriod;
+
+  // how much periods will have in total
+  uint256 public totalPeriods;
+
+  // the amount of tokens to be vested in total
+  uint256 public tokens;
+
+  // how much tokens were claimed
+  uint256 public claimed;
+
+  // the token contract
+  LifToken public token;
+
+  // how much cliff period teh vesting schedule will have
+  uint256 public cliffPeriods;
+
+  // if the vested payment was funded or not
+  bool public funded = false;
+
+  function VestedPayment(
+    uint256 _startTime, uint256 _timePerPeriod,
+    uint256 _totalPeriods, uint256 _cliffPeriods, address tokenAddress
+  ){
+    require(_startTime >= block.timestamp);
+    require(_timePerPeriod > 0);
+    require(_totalPeriods > 0);
+    require(tokenAddress != address(0));
+
+    startTime = _startTime;
+    timePerPeriod = _timePerPeriod;
+    totalPeriods = _totalPeriods;
+    cliffPeriods = _cliffPeriods;
+    token = LifToken(tokenAddress);
+  }
+
+  // fund the vesting contract with tokens allowed to be spent by the contract
+  function fund(uint256 _tokens) onlyOwner {
+    assert(!funded);
+    uint256 tokensAllowed = token.allowance(owner, address(this));
+    require(tokensAllowed >= _tokens);
+
+    token.transferFrom(owner, address(this), _tokens);
+    tokens = _tokens;
+    funded = true;
+  }
+
+  // how much tokens are available to be claimed
+  function getAvailableTokens() public constant returns (uint256) {
+    uint256 period = block.timestamp.sub(startTime).div(timePerPeriod);
+
+    if ((period < cliffPeriods) || !funded) {
+      return 0;
+    } else if (period > totalPeriods) {
+      return tokens.sub(claimed);
+    } else if (period == 0) {
+      return tokens.div(totalPeriods).sub(claimed);
+    } else {
+      return tokens.div(totalPeriods).mul(period).sub(claimed);
```

Comment:
I think the `mul(period)` should be `mul(period + 1)` and once that's done the previous if is not needed

Reply:
:+1: